content/src/service/synchronization/batchDeployer.ts

```diff
@@ -109,81 +109,88 @@ export function createBatchDeployerComponent(
       deploymentsMap.set(entity.entityId, newElementInMap)
 
       const metricLabels = { entity_type: entity.entityType }
-      // increment the gauge of enqueued deployments
-      components.metrics.increment('dcl_pending_deployment_gauge', metricLabels)
 
       const operationPriority = priorityBasedOnEntityType(entity.entityType)
 
-      parallelDeploymentJobs
-        .scheduleJobWithPriority(async () => {
-          /**
-           *  Entity should be marked as processed in the snapshot if anyone of these conditions is met:
-           *  1. The entity is already deployed.
-           *  2. The entity was sucessfully deployed.
-           *  3. The entity failed to be deployed but was successfully persisted as failed deployment
-           */
-          // 1. The entity is already deployed, early return.
-          if (await isEntityDeployed(components, entity.entityId)) {
-            const markAsDeployedFns = deploymentsMap.get(entity.entityId)?.markAsDeployedFns ?? []
-            for (const markAsDeployed of markAsDeployedFns) {
-              await markAsDeployed()
-            }
-            successfulDeployments.add(entity.entityId)
-            deploymentsMap.delete(entity.entityId)
-            return
-          }
-
-          // 2. and 3. We try to deploy the entity or add it to fail deployments
-          let wasEntityProcessed = false
-          let elementInMap = deploymentsMap.get(entity.entityId)
-          if (elementInMap) {
-            try {
-              await deployEntityFromRemoteServer(
-                components,
-                entity.entityId,
-                entity.entityType,
-                entity.authChain,
-                elementInMap.servers,
-                DeploymentContext.SYNCED
-              )
-              wasEntityProcessed = true
+      try {
+        await parallelDeploymentJobs.onSizeLessThan(1000)
```

Comment:
I think this condition will render `operationPriority` useless, should this condition be "priority aware"?
```suggestion
        await parallelDeploymentJobs.onSizeLessThan(1000, operationPriority)
```

Reply:
I don't think so, this is just to put a bound on it, the 1000 elements have their priority.